### ISSUE TITLE
Fabricator Plugins

### ISSUE THREAD

**Issue by jonorherrington** [fbrctr/fabricator#164]
Hey @LukeAskew -

Has there been any thoughts put into adding the ability to make plugins for fabricator that we might not want to have in core, but might want in different circumstances?

Just wanted to spit ball that idea out.

Thanks!

**Comment by LukeAskew** [comment]
Yeah definitely been kicked around. Not sure what it looks like, but I'm open to a brainstorm.

What kind of things is everyone thinking?

**Comment by jameswragg** [comment]
When you say plugin do you mean a JavaScript utility like moment.js or pikaday.js that's used in this toolkit but not the next? Or a Fabricator plugin like commenting on a component for example?

**Comment by LukeAskew** [comment]
I think the latter.

Working on #165 now, and I'm thinking it is a good candidate for plugin status. Not something everyone wants, but enough people seem to find it useful.

**Comment by jameswragg** [comment]
Great, yes that plugin would be appreciated! We've got a LOT of components & the navigation is getting a little unwieldy in it's current state.

(really enjoying Fabricator!)

**Comment by jameswragg** [comment]
Idea for a Fabricator plugin – Resizable components: every component lives in it's own iframe, with a width resize handle. This would provide RWD testing (iframe hitting breakpoints) per component rather than resizing the whole window/viewport & losing your scroll position.

**Comment by ptibbetts** [comment]
:+1: for https://github.com/fbrctr/fabricator/pull/165 , although I personally think that should be included in core.

Continuing from @jameswragg 's point about resizable components, I really liked [ish](http://bradfrost.com/demo/ish/#m), which is included in [PatternLab](http://demo.patternlab.io/).
